astcontext/query_test.go

```diff
@@ -0,0 +1,70 @@
+package astcontext
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestComment(t *testing.T) {
+	var src = `package main
+
+// Hello
+
+/* foo
+bar */
+
+// ..
+// ..
+`
+	opts := &ParserOptions{Src: []byte(src), Comments: true}
+	parser, err := NewParser(opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []struct {
+		offset  int
+		want    Comment
+		wantErr string
+	}{
+		{4, Comment{}, "no comment block"},
+		{9000, Comment{}, "no comment block"},
+		{18, Comment{3, 1, 3, 9}, ""},
+		{24, Comment{5, 1, 6, 7}, ""},
+		{39, Comment{8, 1, 9, 6}, ""},
+	}
+
+	for _, tc := range cases {
+		t.Run(fmt.Sprintf("%v", tc.offset), func(t *testing.T) {
+			out, err := parser.Run(&Query{Mode: "comment", Offset: tc.offset})
+			if !errorContains(err, tc.wantErr) {
+				t.Fatalf("wrong error:\nwant: %v\ngot:  %v", tc.wantErr, err)
+			}
+
+			if err != nil {
+				return
```

Comment:
Can you fail here with `t.Fatalf` ?

Reply:
The error is already checked above. This is just to prevent a nil pointer reference in `out.Comment` below in the tests where we expect `err` to be non-`nil`.